src/test/map-factory-test.js

```diff
@@ -890,4 +914,98 @@ group("The or() method", () => {
     done();
 
   });
+
 });
+
+// group("array edge case", () => {
+
+//   lab.test("A deep array maps properly", done => {
+
+//     const source = {
+//       array: [{ id: 1 }, { id: 2 }, { id: 3 }]
+//     };
+
+//     const expected = {
+//       array: {
+//         levels: [{ id: 1 }, { id: 2 }, { id: 3 }]
+//       }
+//     };
+
+//     const map = createMapper();
+
+//     map("array.[].id").to("array.levels.[].id");
+
+//     const actual = map.execute(source);
+
+//     expect(actual).to.equal(expected);
+
+//     return done();
+
+//   });
+
+//   lab.test("A null source should not do funky stuff when an array is in the mix", done => {
+
+//     const source = {
+//       array: null
+//     };
+
+//     const expected = {
+//     };
+
+//     const map = createMapper();
+
+//     map(["array.[].id"]).to("array.levels.[].id", value => value);
+
+//     const actual = map.execute(source);
+
+//     expect(actual).to.equal(expected);
+
+//     return done();
+
+//   });
+
+//   lab.test("A undefined array in source should not do funky stuff when an array is in the mix", done => {
+
+//     const source = {
+//     };
+
+//     const expected = {
+//     };
+
+//     const map = createMapper();
+
+//     map(["array.[].id"]).to("array.levels.[].id", value => value);
+
+//     const actual = map.execute(source);
+
+//     expect(actual).to.equal(expected);
+
+//     return done();
+
+//   });
+
+//   lab.test("A undefined array in source should not do funky stuff when an array is in the mix", done => {
+
+//     const source = {
+//       "something": {
+//         "array": undefined,
+//         "id": "123"
+//       }
+//     };
+
+//     const expected = {
+//     };
+
+//     const map = createMapper();
+
+//     map(["something.array.[].id"]).to("something.array.levels.[].id", value => value);
+
+//     const actual = map.execute(source);
+
+//     expect(actual).to.equal(expected);
+
+//     return done();
+
+//   });
+
+// });
```

Comment:
More than a thousand lines. Would be great to break it down in a reasonable way...